src/application/Explorer/graphiQLRelay.ts

```diff
@@ -0,0 +1,55 @@
+import Relay from '../../Relay';
+import { GraphiQLResponse, QueryResult } from '../../types';
+import { GRAPHIQL_REQUEST, GRAPHIQL_RESPONSE } from '../../extension/constants';
+
+const graphiQL = new Relay();
+
+graphiQL.listen<GraphiQLResponse>(GRAPHIQL_RESPONSE, ({ payload }) => {
+  if (payload) {
+    graphiQL.broadcast({
+      message: `graphiql:response:${payload.operationName}`,
+      payload: payload.response,
+    });
+  }
+});
+
+export const listenForResponse = (operationName: string, cb: (p) => void) => {
+  const removeListener = graphiQL.listen<QueryResult>(`graphiql:response:${operationName}`, ({ payload }) => {
+    cb(payload);
+    removeListener();
+  });
+};
+
+export const sendGraphiQLRequest = operation => {
+  (window as any).dispatchEvent(new CustomEvent(GRAPHIQL_REQUEST, { 
```

Comment:
You shouldn't need the `as any` on the `window` global references anywhere in this file. Since we've specified the `target` as `es6` in the `tsconfig.json`, the `DOM` library/types are included.